src/fg_apply.rs

```diff
@@ -80,27 +82,16 @@ pub(crate) unsafe fn rav1d_prep_grain<BD: BitDepth>(
     let data = &frame_hdr.film_grain.data;
     let bitdepth_max = (1 << out.p.bpc) - 1;
     let bd = BD::from_c(bitdepth_max);
+    let layout = || r#in.p.layout.try_into().unwrap();
```

Comment:
Why make this a closure instead of just storing it as a variable?

Reply:
There's a branch where it will fail.